src/nfa/concatenation.cc

```diff
@@ -64,7 +64,7 @@ Nfa& Nfa::concatenate(const Nfa& aut) {
 }
 
 Nfa Algorithms::concatenate_eps(const Nfa& lhs, const Nfa& rhs, const Symbol& epsilon, bool use_epsilon,
-                StateToStateMap* lhs_result_states_map, StateToStateMap* rhs_result_states_map) {
+                                StateRenaming* lhs_result_state_renaming, StateRenaming* rhs_result_state_renaming) {
```

Comment:
The names are obviously long, though they are informative, which is the main thing perhaps. I wander whether the _result could be dropped. Not sure about it, as it is now, it really says it all which is nice.

Reply:
This will be modified with output parameters refactorization. We can keep the full names, but have only one output parameter in the function declaration, which will clean up the function declaration nicely. Agreed?

Reply:
if you remove _result ... :)

Reply:
`_result` removed.